### PR TITLE
Accessibility fixes

### LINKED ISSUE #1779: Missing form labels in APD Builder
The APD Builder has a number of form labels missing which will impact our ability to complete 508 testing.

Rationale:

*What It Means*
A form control does not have a corresponding label.
*Why It Matters*
If a form control does not have a properly associated text label, the function or purpose of that form control may not be presented to screen reader users. Form labels also provide visible descriptions and larger clickable targets for form controls.
*How to Fix It*
If a text label for a form control is visible, use the `<label>` element to associate it with its respective form control. If there is no visible label, either provide an associated label, add a descriptive title attribute to the form control, or reference the label(s) using aria-labelledby. Labels are not required for image, submit, reset, button, or hidden form controls.
*The Algorithm... in English*
An `<input>` (except types of image, submit, reset, button, or hidden), `<select>`, or `<textarea>` does not have a properly associated label text. A properly associated label is:
- a `<label>` element with a for attribute value that is equal to the id of a unique form control
- a `<label>` element that surrounds the form control, does not surround any other form controls, and does not reference another element with its for attribute
- a non-empty title attribute, or
- a non-empty aria-labelledby attribute.

### This task is done when...
 - [ ] Each form field in the APD Builder has an appropriate label
 - [ ] WAVE (or comparable testing tool) reports that all form controls have a label

### LINKED ISSUE #1780: Empty table header in APD Builder
Per Wave, there are multiple `<th>` that contains no text. 

Rationale: 

*Why It Matters*
The `<th>` element helps associate table cells with the correct row/column headers. A `<th>` that contains no text may result in cells with missing or incorrect header information.
*How to Fix It*
If the table cell is a header, provide text within the cell that describes the column or row. If the cell is not a header or must remain empty (such as the top-left cell in a data table), make the cell a `<td>` rather than a `<th>`.
*The Algorithm... in English*
A `<th>` element does not contain any text (or contains only spaces) and no images with alternative text.

### This task is done when...
 - [ ] This error is no longer reported in WAVE (or comparable testing device)
 - [ ] Someone else has looked at the code

### PR DESCRIPTION
**Pull request by mgwalker**
### This pull request changes...

- fixes the form label for the state name drop down in the APD state profile section
- fixes a few tables where table header cells were empty by converting them to data cells and styling them like header cells
- closes #1779
- closes #1780

### This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [x] Changelog is updated as appropriate

### This feature is done when...

- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/containers/BudgetSummary.js [hunk 1]
```diff
@@ -15,22 +15,20 @@ const categoryLookup = {
...
+      className={
+        title === categoryLookup.combined
+          ? 'budget-table--subtotal budget-table--row__highlight'
+          : ''
+      }
     >
       <th scope="row">{title}</th>
-      <td
-        className="budget-table--number"
-      >
```
Comment (mgwalker) [inline web/src/containers/BudgetSummary.js]: I love how prettier just randomly switches between these two...

File: web/src/containers/BudgetSummary.js [hunk 2]
```diff
@@ -146,28 +138,17 @@ const BudgetSummary = ({ activities, data, years }) => (
     ))}
 
     <table className="budget-table">
-      <caption className="ds-h4">
-        Project Activities Totals
-      </caption>
+      <caption className="ds-h4">Project Activities Totals</caption>
       <thead>
         <tr>
-          <th id="summary-budget-null1" />
```
Comment (mgwalker) [inline web/src/containers/BudgetSummary.js]: This one actually changed. All that other stuff is just prettier reformatting lines. :pixar-joy:

File: web/src/containers/ApdPreviousActivityTable.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ const ApdPreviousActivityTable = ({
           </th>
         </tr>
         <tr>
-          <th aria-hidden="true" />
+          <td className="th" aria-hidden="true" />
```
Comment (hbillings) [inline web/src/containers/ApdPreviousActivityTable.js]: ahahahaha *sob*

File: web/src/containers/BudgetSummary.js (hunk 1 above)
Comment (hbillings) [reply web/src/containers/BudgetSummary.js]: It's like a treasure hunt! What code actually changed? You'll just have to keep searching...

File: web/src/containers/ApdPreviousActivityTable.js (hunk 3 above)
Comment (mgwalker) [reply web/src/containers/ApdPreviousActivityTable.js]: I am 100% open to a different class name. 😬

**Pull request merged**